### PR TITLE
chore(docs): updating Pokeshop documentation

### PR DESCRIPTION
**Pull request by danielbdias**
This PR aims to update our Pokeshop documentation, giving more context to newcomers that see examples on this API.

To test the documentation, go to `./docs` folder and run `npm install && npm start`.

## Changes

- Changes Pokeshop documentation

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

### CODE REVIEW

File: docs/docusaurus.config.js [hunk 1]
```diff
@@ -136,6 +136,10 @@ const config = {
             to: '/ci-cd-automation/github-actions-pipeline',
             from: ['/ci-cd-automation/ci-cd-best-practices' /*, '/docs/legacyDocFrom2016'*/],
           },
+          {
+            to: '/live-examples/pokeshop',
+            from: ['/pokeshop', '/pokeshop/' /*, '/docs/legacyDocFrom2016'*/],
```
Comment (danielbdias) [inline docs/docusaurus.config.js:141]: This redirect is not working right now, I'll investigate more later to understand what happened.

File: docs/docs/live-examples/pokeshop/list-pokemon.md [hunk 2]
```diff
@@ -1,56 +1,36 @@
...
-Route: `/pokemon?take=20&skip=0`
-
-Method: `GET`
+Route: `GET /pokemon?take=20&skip=0`
 
 Response:
-
-```java
-
+```javascript
```
Comment (mathnogueira) [inline docs/docs/live-examples/pokeshop/list-pokemon.md:12]: Why not `json`?

Comment (danielbdias) [reply docs/docs/live-examples/pokeshop/list-pokemon.md]: Good point, I'll change it to `json`.

**Comment by danielbdias** [comment]
> @danielbdias is this ready to be reviewed?

Not yet. I'm finishing some changes tonight and I believe that I will be release it for you tomorrow morning.

File: docs/docs/live-examples/pokeshop/overview.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# Pokeshop API
+
+As a testing ground, the team at Tracetest has implemented a sample instrumented API around the [PokeAPI](https://pokeapi.co/).
+
+The idea is to have a microservice-divided system that could behave like a typical scenario by having async processes ([RabbitMQ](https://www.rabbitmq.com/)), cache layers ([Redis](https://redis.io/)), database storage ([Postgres](https://www.postgresql.org/)), and simple CRUD interfaces for Pokemons.
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/overview.md:5]:
```suggestion
The idea is to have a microservice-divided system that behaves like a typical scenario by having async processes ([RabbitMQ](https://www.rabbitmq.com/)), cache layers ([Redis](https://redis.io/)), database storage ([Postgres](https://www.postgresql.org/)), and simple CRUD interfaces for Pokemons.
```

File: docs/docs/live-examples/pokeshop/overview.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+With this, users can get familiar with the Tracetest tool by focusing on creating assertions, visualizing the trace, and identifying the different data that comes from the Collector ([Jaeger](https://www.jaegertracing.io/)). Users will learn about basic instrumentation practices like what tools to use, what data to send, when, and what suggested standards need to be followed.
+
+- **Source Code**: https://github.com/kubeshop/pokeshop
+- **Running it locally**: [instructions](https://github.com/kubeshop/pokeshop/blob/master/docs/installing.md#run-it-locally)
+- **Running on Kubernetes**: [instructions](https://github.com/kubeshop/pokeshop/blob/master/docs/installing.md#run-on-a-kubernetes-cluster)
+
+## Use cases
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/overview.md:13]:
```suggestion
## Use Cases
```

File: docs/docs/live-examples/pokeshop/overview.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+We have three use cases that use each component of this structure and that can be observed via Open Telemetry and tested with Tracetest. Each one is triggered by an API call to their respective endpoint:
+
+- [Add Pokemon](./use-cases/add-pokemon.md): add a new pokemon only relying on user input into the database
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/overview.md:17]:
```suggestion
- [Add Pokemon](./use-cases/add-pokemon.md): Add a new Pokemon only relying on user input into the database.
```

File: docs/docs/live-examples/pokeshop/overview.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+- [List Pokemon](./use-cases/list-pokemon.md): lists all Pokemons registered into Pokeshop
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/overview.md:18]:
```suggestion
- [List Pokemon](./use-cases/list-pokemon.md): Lists all Pokemons registered into Pokeshop.
```

File: docs/docs/live-examples/pokeshop/overview.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+- [Import Pokemon](./use-cases/import-pokemon.md): given a Pokemon ID, this endpoint does an async process, going to PokeAPI to get Pokemon data and adding it to the database
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/overview.md:19]:
```suggestion
- [Import Pokemon](./use-cases/import-pokemon.md): Given a Pokemon ID, this endpoint does an async process, going to PokeAPI to get Pokemon data and adding it to the database.
```

File: docs/docs/live-examples/pokeshop/overview.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+## System architecture
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/overview.md:21]:
```suggestion
## System Architecture
```

File: docs/docs/live-examples/pokeshop/overview.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    C --> |IORedis| A
+    C --> |Sequelize| B
+    C --> |ampqlib| D
+    D --> |ampqlib| E
+    E --> |Sequelize| B
+    C --> |OpenTelemetry Node.js SDK| F
+    E --> |OpenTelemetry Node.js SDK| F
+```
+
+In our live tests, we are deploying it into a single Kubernetes namespace, deployed via a [helm chart](https://github.com/kubeshop/pokeshop/blob/master/docs/installing.md#run-on-a-kubernetes-cluster).
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/overview.md:49]:
```suggestion
In our live tests, we are deploying into a single Kubernetes namespace, deployed via a [Helm chart](https://github.com/kubeshop/pokeshop/blob/master/docs/installing.md#run-on-a-kubernetes-cluster).
```

File: docs/docs/live-examples/pokeshop/overview.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+The Pokeshop API is only accessible from within the Kubernetes cluster network as the Tracetest needs to be able to reach it.
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/overview.md:51]:
```suggestion
The Pokeshop API is only accessible from within the Kubernetes cluster network as Tracetest needs to be able to reach it.
```

File: docs/docs/live-examples/pokeshop/use-cases/add-pokemon.md [hunk 11]
```diff
@@ -0,0 +1,97 @@
+# Pokeshop API - Add Pokemon
+
+This use case showcases a simple example where a user makes an API call with some data and this data is validated and persisted into a database.
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/add-pokemon.md:3]:
```suggestion
This use case showcases a simple example of an API call where call data is validated and persisted into a database.
```

File: docs/docs/live-examples/pokeshop/use-cases/add-pokemon.md [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+{
+  "id":  1000,
+  "name":  "meowth",
+  "type":  "normal",
+  "imageUrl":  "https://assets.pokemon.com/assets/cms2/img/pokedex/full/052.png",
+  "isFeatured":  true
+}
+```
+
+## Building a test for this scenario
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/add-pokemon.md:44]:
```suggestion
## Building a Test for This Scenario
```

File: docs/docs/live-examples/pokeshop/use-cases/add-pokemon.md [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+
+Using Tracetest, we can [create a test](../../../web-ui/creating-tests.md) that will execute an API call on `POST /pokemon` and validate two properties:
+- The API should return a proper result with HTTP 201 Created
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/add-pokemon.md:47]:
```suggestion
- The API should return a proper result with **HTTP 201 Created**.
```

File: docs/docs/live-examples/pokeshop/use-cases/add-pokemon.md [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+- The database should return with low latency (< 200ms)
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/add-pokemon.md:48]:
```suggestion
- The database should return with **low latency (< 200ms)**.
```

File: docs/docs/live-examples/pokeshop/use-cases/add-pokemon.md [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+
+With this trace, now we can build [assertions](../../../concepts/assertions.md) on Tracetest and validate the API response and the database latency:
+
+- **The API should return a proper result with HTTP 201 Created:**
+![](../images/add-pokemon-api-test-spec.png)
+
+- **The database should return with low latency (< 200ms):**
+![](../images/add-pokemon-database-test-spec.png)
+
+And that's all, now you can validate this entire use case.
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/add-pokemon.md:65]:
```suggestion
Now you can validate this entire use case.
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 16]
```diff
@@ -0,0 +1,153 @@
...
+      Database-->>API: pokemon
+
+      API->>Cache: populate cache
+      Cache-->>API: ok
+
+      API-->>Endpoint: 200 OK <br> <Pokemon object>
+    end
+```
+
+You can trigger this use case by calling the endpoint `GET /pokemon/25` without payload, and should receive a payload similar to this:
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:30]:
```suggestion
You can trigger this use case by calling the endpoint `GET /pokemon/25` without payload and should receive a payload similar to this: 
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 17]
```diff
@@ -0,0 +1,153 @@
...
+{
+  "id":  25,
+  "name":  "pikachu",
+  "type":  "electric",
+  "imageUrl":  "https://assets.pokemon.com/assets/cms2/img/pokedex/full/025.png",
+  "isFeatured":  true
+}
+```
+
+## Building a test for those scenarios
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:41]:
```suggestion
## Building a Test for the Described Scenarios
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 18]
```diff
@@ -0,0 +1,153 @@
...
+
+Using Tracetest, we can [create two tests](../../../web-ui/creating-tests.md) that will execute an API call on `GET /pokemon/25` and validate the following scenarios:
+1. **an API call with a cache hit**
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:44]:
```suggestion
1. **An API call with a cache hit.**
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 19]
```diff
@@ -0,0 +1,153 @@
...
+   - The API should return a valid result with HTTP 200 OK
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:45]:
```suggestion
   - The API should return a valid result with HTTP 200 OK.
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 20]
```diff
@@ -0,0 +1,153 @@
...
+   - The cache should be queried
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:46]:
```suggestion
   - The cache should be queried.
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 21]
```diff
@@ -0,0 +1,153 @@
...
+   - The database should not be queried
+2. **an API call with a cache miss**
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:48]:
```suggestion
2. **An API call with a cache miss.**
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 22]
```diff
@@ -0,0 +1,153 @@
...
+   - The API should return a valid result with HTTP 200 OK
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:49]:
```suggestion
   - The API should return a valid result with HTTP 200 OK.
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 23]
```diff
@@ -0,0 +1,153 @@
...
+   - The cache should be queried
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:50]:
```suggestion
   - The cache should be queried.
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 24]
```diff
@@ -0,0 +1,153 @@
...
+
+Using Tracetest, we can [create two tests](../../../web-ui/creating-tests.md) that will execute an API call on `GET /pokemon/25` and validate the following scenarios:
+1. **an API call with a cache hit**
+   - The API should return a valid result with HTTP 200 OK
+   - The cache should be queried
+   - The database should not be queried
+2. **an API call with a cache miss**
+   - The API should return a valid result with HTTP 200 OK
+   - The cache should be queried 
+   - The cache should be populated
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:51]:
```suggestion
   - The cache should be populated.
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 25]
```diff
@@ -0,0 +1,153 @@
...
+   - The database should be queried
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:52]:
```suggestion
   - The database should be queried.
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 26]
```diff
@@ -0,0 +1,153 @@
...
+
+### Traces
+
+Running these tests for the first time will create an Observability trace with two different shapes, depending on the cache situation.
+
+1. **cache miss**, where we can see spans from the API, database, and cache:
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:58]:
```suggestion
1. **Cache Miss** where we can see spans from the API, database, and cache:
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 27]
```diff
@@ -0,0 +1,153 @@
...
+![](../images/get-pokemon-by-id-trace-cachemiss.png)
+
+2. **cache hit**, where we can see spans from the API and cache:
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:61]:
```suggestion
2. **Cache Hit** where we can see spans from the API and cache:
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 28]
```diff
@@ -0,0 +1,153 @@
...
+![](../images/get-pokemon-by-id-trace-cachehit.png)
+
+### Assertions
+
+With this trace, now we can build [assertions](../../../concepts/assertions.md) on Tracetest and validate API, cache, and database responses:
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:66]:
```suggestion
With this trace, we can build [assertions](../../../concepts/assertions.md) on Tracetest and validate API, cache, and database responses:
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 29]
```diff
@@ -0,0 +1,153 @@
...
+
+- [both cases] The API should return a valid result with HTTP 200 OK
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:68]:
```suggestion
- [both cases] The API should return a valid result with HTTP 200 OK.
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 30]
```diff
@@ -0,0 +1,153 @@
...
+![](../images/get-pokemon-by-id-api-test-spec.png)
+
+- [both cases] The cache should be queried
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:71]:
```suggestion
- [both cases] The cache should be queried.
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 31]
```diff
@@ -0,0 +1,153 @@
...
+![](../images/get-pokemon-by-id-redis-query-test-spec.png)
+
+- [cache hit] The database should not be queried
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:74]:
```suggestion
- [cache hit] The database should not be queried.
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 32]
```diff
@@ -0,0 +1,153 @@
...
+![](../images/get-pokemon-by-id-db-no-query-test-spec.png)
+
+- [cache miss] The cache should be populated
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:77]:
```suggestion
- [Cache Miss] The cache should be populated.
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 33]
```diff
@@ -0,0 +1,153 @@
...
+![](../images/get-pokemon-by-id-redis-set-test-spec.png)
+
+- [cache miss] The database should be queried
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:80]:
```suggestion
- [Cache Miss] The database should be queried.
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 34]
```diff
@@ -0,0 +1,153 @@
...
+![](../images/get-pokemon-by-id-db-query-test-spec.png)
+
+And that's all, now you can validate this entire use case.
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:83]:
```suggestion
Now you can validate this entire use case.
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 35]
```diff
@@ -0,0 +1,153 @@
...
+
+### Test Definition
+
+If you want to replicate those tests on Tracetest see by yourself, you can replicate these steps on our Web UI or using our CLI, saving one of the test definitions as the file `test-definition.yml` and running:
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:87]:
```suggestion
If you want to replicate those tests on Tracetest, you can replicate these steps on our Web UI or using our CLI, saving one of the test definitions as the file `test-definition.yml` and running:
```

File: docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md [hunk 36]
```diff
@@ -0,0 +1,153 @@
...
+
+```sh
+tracetest test -d test-definition.yml --wait-for-results
+```
+
+#### Cache miss scenario
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/get-pokemon-by-id.md:93]:
```suggestion
#### Cache Miss Scenario
```

File: docs/docs/live-examples/pokeshop/use-cases/import-pokemon.md [hunk 37]
```diff
@@ -0,0 +1,138 @@
+# Pokeshop API - Import Pokemon
+
+This use case showcases a more complex scenario involving an async process. Usually, when working with microservices, there are use cases where some of the processing needs to happen asynchronously, for example, when triggering a user notification, generating reports, or processing a payment order. With this endpoint, we provide an example of how users can implement trace-based testing for such scenarios.
+
+Here the process in split in two phases:
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/import-pokemon.md:5]:
```suggestion
Here the process is split into two phases: 
```

File: docs/docs/live-examples/pokeshop/use-cases/import-pokemon.md [hunk 38]
```diff
@@ -0,0 +1,138 @@
+# Pokeshop API - Import Pokemon
+
+This use case showcases a more complex scenario involving an async process. Usually, when working with microservices, there are use cases where some of the processing needs to happen asynchronously, for example, when triggering a user notification, generating reports, or processing a payment order. With this endpoint, we provide an example of how users can implement trace-based testing for such scenarios.
+
+Here the process in split in two phases: 
+1. an API call that enqueue a import request to a queue
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/import-pokemon.md:6]:
```suggestion
1. An API call that enqueues an import request to a queue.
```

File: docs/docs/live-examples/pokeshop/use-cases/import-pokemon.md [hunk 39]
```diff
@@ -0,0 +1,138 @@
...
+```
+
+It should return the following payload:
+```json
+{
+  "id":  52
+}
+```
+
+## Building a test for this scenario
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/import-pokemon.md:62]:
```suggestion
## Building a Test for This Scenario
```

File: docs/docs/live-examples/pokeshop/use-cases/import-pokemon.md [hunk 40]
```diff
@@ -0,0 +1,138 @@
...
+
+Using Tracetest, we can [create a test](../../../web-ui/creating-tests.md) that will execute an API call on `POST /pokemon/import` and validate the following properties:
+- The API should enqueue an import task and return HTTP 200 OK
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/import-pokemon.md:65]:
```suggestion
- The API should enqueue an import task and return HTTP 200 OK.
```

File: docs/docs/live-examples/pokeshop/use-cases/import-pokemon.md [hunk 41]
```diff
@@ -0,0 +1,138 @@
...
+- The worker should dequeue the import task
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/import-pokemon.md:66]:
```suggestion
- The worker should dequeue the import task.
```

File: docs/docs/live-examples/pokeshop/use-cases/import-pokemon.md [hunk 42]
```diff
@@ -0,0 +1,138 @@
...
+- PokeAPI should return a valid response
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/import-pokemon.md:67]:
```suggestion
- PokeAPI should return a valid response.
```

File: docs/docs/live-examples/pokeshop/use-cases/import-pokemon.md [hunk 43]
```diff
@@ -0,0 +1,138 @@
...
+
+### Traces
+
+Running these tests for the first time will create an Observability trace like the image above, where you can see spans for the API call, the queue messaging, the PokeAPI (external API) call, and database calls. One interesting thing about this trace is that **you can observe the entire use case, end to end**:
+
+![](../images/import-pokemon-trace.png)
+
+### Assertions
+
+With this trace, now we can build [assertions](../../../concepts/assertions.md) on Tracetest and validate the API and Worker behaviors:
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/import-pokemon.md:78]:
```suggestion
With this trace, we can build [assertions](../../../concepts/assertions.md) on Tracetest and validate the API and Worker behaviors:
```

File: docs/docs/live-examples/pokeshop/use-cases/list-pokemon.md [hunk 44]
```diff
@@ -0,0 +1,100 @@
+# Pokeshop API - List Pokemon
+
+This use case retrieves the list of Pokemon directly from the database (Postgres) based on the provided query through the API. The idea of this query is to showcase a straightforward scenario, where the API layer receives a request from the outside and needs to trigger a database query to get some data and return it to the client.
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/list-pokemon.md:3]:
```suggestion
This use case retrieves the list of Pokemon directly from the database (Postgres) based on the provided query through the API. The purpose of this query is to showcase a straightforward scenario, where the API layer receives a request from the outside and needs to trigger a database query to get some data and return it to the client.
```

File: docs/docs/live-examples/pokeshop/use-cases/list-pokemon.md [hunk 45]
```diff
@@ -0,0 +1,100 @@
...
+    
+    Endpoint->>API: request
+
+    API->>Database: get list of pokemons
+    Database-->>API: list of pokemons
+
+    API-->>Endpoint: 200 OK <br> [<Pokemon object>, ...]
+```
+
+You can trigger this use case by calling the endpoint `GET /pokemon?take=20&skip=0` without payload, and should receive a payload similar to this:
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/list-pokemon.md:19]:
```suggestion
You can trigger this use case by calling the endpoint `GET /pokemon?take=20&skip=0` without a payload and should receive a payload similar to this: 
```

File: docs/docs/live-examples/pokeshop/use-cases/list-pokemon.md [hunk 46]
```diff
@@ -0,0 +1,100 @@
...
+    "name":  "raichu",
+    "type":  "electric",
+    "imageUrl":  "https://assets.pokemon.com/assets/cms2/img/pokedex/full/026.png",
+    "isFeatured":  true
+  },
+  //...
+]
+```
+
+## Building a test for this scenario
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/list-pokemon.md:40]:
```suggestion
## Building a Test for This Scenario
```

File: docs/docs/live-examples/pokeshop/use-cases/list-pokemon.md [hunk 47]
```diff
@@ -0,0 +1,100 @@
...
+
+Using Tracetest, we can [create a test](../../../web-ui/creating-tests.md) that will execute an API call on `GET /pokemon` and validate three properties:
+- The API should return results with HTTP 200 OK
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/list-pokemon.md:43]:
```suggestion
- The API should return results with HTTP 200 OK.
```

File: docs/docs/live-examples/pokeshop/use-cases/list-pokemon.md [hunk 48]
```diff
@@ -0,0 +1,100 @@
...
+- The database should respond with low latency (< 200ms)
+- The database query should use the query string parameters `take` and `skip` correctly
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/list-pokemon.md:45]:
```suggestion
- The database query should use the query string parameters `take` and `skip` correctly.
```

File: docs/docs/live-examples/pokeshop/use-cases/list-pokemon.md [hunk 49]
```diff
@@ -0,0 +1,100 @@
...
+
+### Traces
+
+Running these tests for the first time will create an Observability trace like the image above, where you can see spans for the API call and database calls:
+![](../images/list-pokemons-trace.png)
+
+### Assertions
+
+With this trace, now we can build [assertions](../../../concepts/assertions.md) on Tracetest and validate the API response and the database responses:
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/list-pokemon.md:54]:
```suggestion
With this trace, we can build [assertions](../../../concepts/assertions.md) on Tracetest and validate the API response and the database responses:
```

File: docs/docs/live-examples/pokeshop/use-cases/list-pokemon.md [hunk 50]
```diff
@@ -0,0 +1,100 @@
...
+- **The API should return results with HTTP 200 OK:**
+![](../images/list-pokemons-api-test-spec.png)
+
+- **The database should respond with low latency (< 200ms):**
+![](../images/list-pokemons-db-latency-test-spec.png)
+
+- **The database query should use the query string parameters `take` and `skip` correctly:**
+![](../images/list-pokemons-db-query-test-spec.png)
+
+And that's all, now you can validate this entire use case.
```
Comment (jfermi) [inline docs/docs/live-examples/pokeshop/use-cases/list-pokemon.md:65]:
```suggestion
Now you can validate this entire use case.
```

File: docs/docs/live-examples/pokeshop/overview.md [hunk 51]
```diff
@@ -0,0 +1,51 @@
...
+- **Running it locally**: [instructions](https://github.com/kubeshop/pokeshop/blob/master/docs/installing.md#run-it-locally)
+- **Running on Kubernetes**: [instructions](https://github.com/kubeshop/pokeshop/blob/master/docs/installing.md#run-on-a-kubernetes-cluster)
+
+## Use Cases
+
+We have three use cases that use each component of this structure and that can be observed via Open Telemetry and tested with Tracetest. Each one is triggered by an API call to their respective endpoint:
+
+- [Add Pokemon](./use-cases/add-pokemon.md): Add a new Pokemon only relying on user input into the database.
+- [List Pokemon](./use-cases/list-pokemon.md): Lists all Pokemons registered into Pokeshop.
+- [Import Pokemon](./use-cases/import-pokemon.md): Given a Pokemon ID, this endpoint does an async process, going to PokeAPI to get Pokemon data and adding it to the database.
```
Comment (xoscar) [inline docs/docs/live-examples/pokeshop/overview.md:19]: I think we can add the rest here, like the featured endpoint or the newly created get pokemon by id

**Review by xoscar — APPROVED** [review]
Really well done @danielbdias! 🔥

**Pull request merged**